pvlib/tests/spectrum/test_mismatch.py

```diff
@@ -1,11 +1,10 @@
 import pytest
-from numpy.testing import assert_allclose, assert_approx_equal, assert_equal
+from numpy.testing import assert_allclose, assert_approx_equal
 import pandas as pd
 import numpy as np
 from pvlib import spectrum
-from pvlib._deprecation import pvlibDeprecationWarning
 
-from .conftest import DATA_DIR, assert_series_equal, fail_on_pvlib_version
+from ..conftest import DATA_DIR, assert_series_equal
 
 SPECTRL2_TEST_DATA = DATA_DIR / 'spectrl2_example_spectra.csv'
```

Comment:
This variable, and the `spectrl2_data` fixture, can be removed from this file.  They're only needed in the new `test_spectrl2.py`.

Reply:
These parts are still required in `test_mismatch.py` because the first test for `calc_mismatch_field` uses spectlr2 data

Reply:
Indeed so!  My mistake.

Reply:
In that case, can we extract that fixture out into a `conftest.py` file inside this new `pvlib/tests/spectrum` directory?  See https://docs.pytest.org/en/stable/reference/fixtures.html#conftest-py-sharing-fixtures-across-multiple-files

Then we do not have to have redundant definitions in these two files.

Reply:
Nice idea. I had been wondering if this was possible somehow but was not really aware of how `conftest.py` works to be able to suggest this. Commit [3729860](https://github.com/pvlib/pvlib-python/pull/2151/commits/3729860d35f8da81b18be92c5eefb8b7b86a3d5f) should resolve this.